stylix/mk-target.nix

```
@@ -0,0 +1,73 @@
+{
+  # name used for creation of options
+  name,
+
+  # name used for description of options
+  humanName,
+
+  # condition for automatically enabling this target
+  autoEnable ? true,
+
+  # options to be added under `options.stylix.targets.${name}`
+  extraOptions ? { },
+
+  # list of functions with args representing Stylix options; function results
+  # only merged if all relevant Stylix options are not null
+  configElements ? [ ],
+
+  # function with args representing Stylix options; function result always
+  # merged
+  generalConfig ? null,
+}:
+{ config, lib, ... }:
+let
+  cfg = config.stylix.targets.${name};
+in
+{
+  imports = [
+    {
+      options.stylix.targets.${name}.enable =
+        config.lib.stylix.mkEnableTarget humanName autoEnable;
+    }
+    {
+      options.stylix.targets.${name} = extraOptions;
+    }
+  ];
+
+  config =
+    let
+      functionArgNames =
+        fn:
+        lib.pipe fn [
+          lib.functionArgs
+          builtins.attrNames
+        ];
+      getStylixAttrs =
+        fn:
+        lib.genAttrs (functionArgNames fn) (
+          arg:
+          if arg == "cfg" then
+            cfg
+          else if arg == "colors" then
+            config.lib.stylix.${arg}
+          else
+            config.stylix.${arg} or throw
+              "stylix: Stylix arg was requested by mk-target but does not exist: ${arg}"
+        );
+      mkConfig = fn: fn (getStylixAttrs fn);
+      mkConditionalConfig =
+        fn:
+        let
+          attrs = functionArgNames fn;
+        in
+        lib.mkIf (builtins.all (
+          attr: if attr == "cfg" then true else attr != null
```

I think this is the same:

```suggestion
          attr: attr == "cfg" || attr != null
```

Or maybe even this? :eyes: 

```suggestion
          attr: attr != null
```

Since `attr == "cfg"` is also a non-null attr.